### ISSUE TITLE
Elastic Load Balancer does not support sc_apikey header

### ISSUE THREAD

**Issue by thaiphan** [Sitecore/jss#1060]
## Description

I'm trying to use the `GraphQLSitemapService`. However, I'm getting the following error:

```
SSC API key is required. Pass with 'sc_apikey' query string or HTTP header.
```

It turns out that Elastic Load Balancer strips out non-standard custom headers.

## Possible Fix

It looks like `GraphQLRequestClient` passes  `sc_apikey` as a HTTP header. Is it possible to switch it so that it's a query param? Or configure it so we can choose one or the other...

https://github.com/Sitecore/jss/blob/f1fd3f1e17906162c70c055d94b4cff70daa88df/packages/sitecore-jss/src/graphql-request-client.ts#L52

**Comment by Elyseum** [comment]
Similar issue for running Sitecore in K8s behind nginx (underscore is not allowed by default, the issue + fix is documented in Sitecore 10.2 installation PDF).

Nick Wesselman worked around this by modifying his generate-config file. It includes the sc_apikey as query string parameter in the GraphQL URL. See https://github.com/Sitecore/Helix.Examples/blob/master/examples/helix-basic-nextjs/src/Project/BasicCompany/nextjs/scripts/generate-config.ts

**Comment by ambrauer** [comment]
This was switched to a header by default with the introduction of Experience Edge (JSS 18, see #636) as a [header is required here](https://doc.sitecore.com/xp/en/developers/hd/200/sitecore-headless-development/create-the-api-key.html#:~:text=Sitecore%20Services%20Client%20and%20Sitecore%20GraphQL%20also%20accept%20API%20keys%20via%20query%20string.%20However%2C%20we%20recommend%20to%20use%20the%20sc_apikey%20HTTP%20header%20as%20this%20is%20mirrors%20the%20behavior%20of%20Experience%20Edge.). 

You can use a query string by modifying the generate-config.ts in your app as suggested above (here's the [pre-edge version](https://github.com/Sitecore/jss/blob/release/16.0.0/samples/nextjs/scripts/generate-config.ts#L38-L39), which is what the referenced Helix.Examples is using).
